Sources/ServiceLifecycle/GracefulShutdown.swift

```diff
@@ -55,6 +55,18 @@ public func withGracefulShutdownHandler<T>(
     return try await operation()
 }
 
+/// Suspend the current task until graceful shutdown is executed.
+///
+/// A common use-case is to call this method in the `operation` closure of ``withGracefulShutdownHandler(operation:onGracefulShutdown:)``
+/// when you are only interested in executing a closure upon graceful shutdown, and nothing else.
+public func waitForGracefulShutdown() async {
+    guard let gracefulShutdownManager = TaskLocals.gracefulShutdownManager else {
+        return
+    }
+    
+    await gracefulShutdownManager.waitForGracefulShutdown()
```

Comment:
That's not the right way to do this and it doesn't respect cancellation. We have a `AsyncGracefulShutdownSequence` that we should just use here with `try await AsyncGracefulShutdownSequence().first { _ in }`. 

Furthermore, we should probably change the implementation of that sequence to use an `AsyncThrowingStream` instead so we get the `CancellationError` thrown.